conwayste/src/ui/ui_errors.rs

```diff
@@ -22,7 +22,7 @@ custom_error! {pub UIError
     WidgetNotFound {reason: String} = "UIError::WidgetNotFound({reason})",
     InvalidAction {reason: String} = "UIError::InvalidAction({reason})",
     ActionRestricted{reason: String} = "UIError::ActionRestricted({reason})",
-    WidgetIDCollision{reason: String} = "UIError::WidgetIDCollision({reason})",
+    NodeIDCollision{reason: String} = "UIError::NodeIDCollision({reason})",
```

Comment:
I don't think this error, `NodeIDCollision` is physically possible due to the atomic uniqueness of `NodeID`?? I have no issues keeping it around and checking anyway if you think we should.

Reply:
Love that last idea!! I'd like to incorporate it in this PR. :smile: 

The case I thought of is if we somehow end up with a widget where a widget's node-id is `Some`, and it tries to be reinserted into the tree. Currently when we remove a widget from the tree, it's dropped. We don't return it to the caller. That case might only occur if we did return it to the caller. 

I think I'll leave the error in for now until the code matures a little more.